### ISSUE TITLE
[Bug] excessive CPU on 

### ISSUE THREAD

**Issue by weepingwillowben** [DLR-RM/stable-baselines3#413]
### 🐛 Bug

Excessive CPU utilization with CnnPolicy.

### To Reproduce

On a GCP deep learning instance, the following code snippet uses 8 full CPUs during training. Note that on our private servers, this does not occur. Not clear whether this is due to cloud virtualization or to the more recent version of cuda (11.1) on GCP.

```python
from stable_baselines3 import PPO
from stable_baselines3.common.atari_wrappers import AtariWrapper
import gym
import supersuit as ss

env = gym.make("SpaceInvadersNoFrameskip-v4")
env = AtariWrapper(env)

model = PPO("CnnPolicy", env)
model.learn(total_timesteps=2000000)
model.save("policy")
```

setting `torch.set_num_threads(1)` or `OMP_NUM_THREADS=1` or `MKL_NUM_THREADS=1` all successfully run the code in a single thread.

### Expected behavior

We expect this to be single threaded or to be using a small number of threads when CUDA is enabled. The fact that it doesn't is a bit disturbing, as it suggests that perhaps some operation is running on CPU instead of GPU.

### System Info

Describe the characteristic of your environment:
 * installed version 1.0 with pip
 * cuda 11.0
 * Cudnn 8.0.5
 * pytorch 1.7.1
 * python 3.7

Only seems to happen on a GCP instance

### Checklist

- [ ] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)
- [ ] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [ ] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by Miffyli** [comment]
Have you checked that CUDA is available in PyTorch? SB3 uses `torch.is_cuda_available()` internally to see if it should use GPU or CPU with default parameters (GPU if CUDA is available). If CUDA is not available, CPU is used which could lead to high usage.

Note that some operations are still done on CPU, e.g. data storing/processing and environment itself, so CPU is not completely underutilized.

**Comment by weepingwillowben** [comment]
Yes, CUDA is avaliable, and I have checked that it is indeed using the GPU when running nvidia-smi

**Comment by weepingwillowben** [comment]
Indeed, it runs much faster when using the GPU.

**Comment by Miffyli** [comment]
Quickly running the code, without `import supersuit` and with similar system (e.g. PyTorch 1.8.1), does spike the CPU usage at four cores (our of 8 logical ones) occasionally, which likely corresponds to the training sessions where data is passed around. Sounds like GCP might be measuring/handling things differently, but generally I would not worry about it, especially if `torch.num_threads` fixes the situation (this also makes training faster in most cases).

**Comment by MotorCityCobra** [comment]
`stable-baselines3` uses PyTorch?  
I can't get the A2C algorithm to use my GPU. I cannot set the device to 'cuda'.
Do I need to install PyTorch?

**Comment by Miffyli** [comment]
@MotorCityCobra Yes, SB3 uses PyTorch. Installing SB3 will also install PyTorch if you do not have it. Make sure your PyTorch is installed with CUDA support (see [here](https://pytorch.org/get-started/locally/)) and that you have a Nvidia GPU in your machine.

**Comment by divyanshj16** [comment]
I have PyTorch installed with cuda, but still I can't run it on GPU. On my local machine it runs on gpu, on a AWS instance its only running on CPU. IOs there a solution to this?

**Comment by Miffyli** [comment]
Make sure there is a valid GPU and drivers installed on the AWS instance, and check that `torch.cuda.is_available()` returns `True`. SB3 does use GPU if torch reports that one is available.

**Comment by divyanshj16** [comment]
I checked it `torch.cuda.is_available()`, its True.

**Comment by Miffyli** [comment]
According this output the gpu is being used. Your observations/actions might be so small that GPU is not being excessively utilized. If you use small 1D vectors as observations, I recommend using CPU device and `torch.set_num_threads(1)` to speed up training.

**Comment by divyanshj16** [comment]
My CPU usage is full, but GPU is not being used. Its only happening on AWS. On my local machine its running much faster and also using the GPU. And the input is RGB image.


```python
import torch
print("torch cuda is_avaialble: ", torch.cuda.is_available())
#torch.set_num_threads(1)

import robosuite as suite
from robosuite.wrappers import GymWrapper
import robosuite.utils.macros as macros
macros.IMAGE_CONVENTION = "opencv"
from stable_baselines3 import PPO


image_size = 256

env = GymWrapper(
    suite.make(
            "Lift",
            robots = ["IIWA"],
            reward_shaping=True,
            has_renderer=False,           
            has_offscreen_renderer=True,
            ignore_done=False,
            use_object_obs=True,
            use_camera_obs=True,  
            render_camera='frontview',
            control_freq=20, 
            camera_names=['agentview'],
            camera_heights=image_size,
            camera_widths=image_size,
            render_with_igibson=False
        ),
        keys=['agentview_image']
) 

model = PPO("CnnPolicy", env, verbose=1, device='cuda')
model.learn(total_timesteps=30000)

import imageio
video_writer = imageio.get_writer('video.mp4', fps=30)
obs = env.reset()
for i in range(1000):
    action, _states = model.predict(obs, deterministic=True)
    obs, reward, done, info = env.step(action)
    video_img = obs.transpose(1,2,0)
    video_writer.append_data(video_img)
    if done:
        break

video_writer.close()

env.close()
```


Here is the output of `nvidia-smi`

<img width="722" alt="image" src="https://user-images.githubusercontent.com/16683472/126867219-35505570-e7db-4fe7-acaa-46b1d10e5ee2.png">

**Comment by Miffyli** [comment]
Judging by the memory use and power state (P0), the GPU indeed is being used, but the input sizes are so small it does not get to utilize V100 too much. Does it still run faster than without CUDA on AWS? If so, everything checks out. To make better use of the GPU, you could increase the number of environments, the batch size and `n_steps`.

**Comment by divyanshj16** [comment]
@Miffyli 
Thanks for the pointers. It was indeed an issue with mujoco,it was not using gpu for rendering. After fixing that its much faster, but still my local machine has faster rendering. I am resolving that now.

**Comment by araffin** [comment]
Closing as the issues seems to be related to PyTorch/GCP and was solved by setting `OMP_NUM_THREADS` to a lower value.
